src/data/polynomial/module.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+
+import ring_theory.finiteness
+
+/-!
+# Polynomial module
+
+In this file, we define the polynomial module for an `R`-module `M`, i.e. the `R[X]`-module `M[X]`.
+
+-/
+
+universes u v
+
+open polynomial
+open_locale polynomial big_operators
+
+variables {R M : Type*} [comm_ring R] [add_comm_group M] [module R M] (I : ideal R)
+
+/-- The `R[X]`-module `M[X]` for an `R`-module `M`. -/
+abbreviation polynomial_module (M : Type*) [add_comm_monoid M] := ℕ →₀ M
+
+namespace polynomial_module
+
+-- this is definitionally equal to `module_polynomial_of_endo _`, but that is not used since the
+-- simp lemmas associated to it is undesirable here.
+@[semireducible] noncomputable
+instance : module R[X] (polynomial_module M) :=
+module_polynomial_of_endo ({ to_fun := λ (f : polynomial_module M),
```

Comment:
I don't understand the comment; aren't you exactly using `module_polynomial_of_endo`?

It seems you can use [finsupp.lmap_domain](https://leanprover-community.github.io/mathlib_docs/linear_algebra/finsupp.html#finsupp.lmap_domain) here?

Can you show `M[X]` `is_tensor_product` of `R[X]` and `M` over `R`? Or better, it's isomorphic to R[X] ⊗ₜ[R] M as R[X]-modules (tensor_product.left_module)? Or that it's an extension of scalars (#15673 doesn't introduce such a predicate though)?

Reply:
I think [`is_base_change`](https://leanprover-community.github.io/mathlib_docs/ring_theory/is_tensor_product.html#is_base_change) is what you are looking for?
And it is indeed on my todo list - though it needs a bit more api around `is_base_change`, so it probably won't be in this PR.